joomla/joomla-cms
# Regression: Bring back translateformat feature to JFormFieldCalendar

PR https://github.com/joomla/joomla-cms/pull/11138 removed the new "translateFormat" feature of the calendar formfield which was introduced with https://github.com/joomla/joomla-cms/pull/12102

### Summary of Changes
Brings back the same code.

### Testing Instructions
Testing instructions from original PR apply here as well:

1. Apply PR
2. Add the new language strings to your second language you have installed. Eg for de-DE add this to both de-DE.ini files (frontend and backend):

```
DATE_FORMAT_CALENDAR_DATE="%d.%m.%Y"
DATE_FORMAT_CALENDAR_DATETIME="%d.%m.%Y %H:%M:%S"
DATE_FORMAT_FILTER_DATETIME="d.m.Y H:i:s"
DATE_FORMAT_FILTER_DATE="d.m.Y"
```
1. Check backend article edits and watch the format changing from YYYY-MM-DD to DD.MM.YYYY if you change the language.
2. Active the user - profile plugin and edit your profil in frontend. Check the birthday field and watch the change in format. Also this field should **not** show the time.

### Note about the strings
* `DATE_FORMAT_CALENDAR_DATE` and `DATE_FORMAT_CALENDAR_DATETIME` are the strings used to display of the calendar. They use the formats from PHP strftime (http://php.net/manual/en/function.strftime.php)
* `DATE_FORMAT_FILTER_DATE` and `DATE_FORMAT_FILTER_DATETIME` are the strings used to convert the entered date back to the english format. They use the formats from PHP date (http://php.net/manual/en/function.date.php)
